heketi/heketi
# Backup database to Kubernetes Secret

This change enables Heketi to automatically backup the database to a
Kubernetes secret.  It does this by adding a new middleware which
determines if the database needs to be backed up or not by checking the
request and its result.

With this change, there is no longer a need to deploy Heketi twice or
create a GlusterFS volume to store the database.

A future patch will add functional tests once the functional test
framework is updated.

How to test:
Change the image in extras/kubernetes/heketi-deploy.json to
`lpabon/heketi:secretdb`, and the value of HEKETI_EXECUTOR to `mock`
then deploy.  Insert to fake systems using heketi-cli (you can use the
topology-sample.json file in client/cli/go).  Now edit deployment and
change the value of snapshot limit.  This will cause Kubernetes to
redeploy Heketi.  Check the new Heketi server has the data that was
inserted.

Closes #529
Closes #631

Signed-off-by: Luis Pabón <luis.pabon@coreos.com>
